core/string/ustring.cpp

```diff
@@ -4245,6 +4245,45 @@ String String::dedent() const {
 	return new_string;
 }
 
+String String::strip(const String &p_chars) const {
+	int len = length();
+	int end = len - 1;
+	int beg = 0;
+
+	if (p_chars.is_empty()) {
+		// Strip non-printable characters by default.
+		for (; beg < len; beg++) {
+			if (get(beg) > 32) {
+				break;
+			}
+		}
+		for (; end >= 0; end--) {
+			if (get(end) > 32) {
+				break;
+			}
+		}
+	} else {
+		// Strip given characters.
+		for (; beg < len; beg++) {
+			if (p_chars.find_char(get(beg)) == -1) {
+				break;
+			}
+		}
+
+		for (; end >= 0; end--) {
+			if (p_chars.find_char(get(end)) == -1) {
+				break;
+			}
+		}
+	}
+
+	if (beg == 0 && end == len - 1) {
```

Comment:
This check is unnecessary if you're calling `substr` anyway; it does the same check again.